apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
+
+At present, there are 6 primitive data types.
+New ones cannot be created, although that is a planned extension in the near-future, whereby new ones will be defined by combining the primitive data types and constraints such as max lengths, minimum values, etc. ([see the associated in-development RFC](https://github.com/blockprotocol/blockprotocol/pull/355)).
+
+When represented as JSON schema, the 6 primitive data types are:
+
+#### Text
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1",
+  "title": "Text",
+  "description": "An ordered sequence of characters",
+  "type": "string"
+}
+```
+
+#### Number
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/number/v/1",
+  "title": "Number",
+  "description": "An arithmetical value (in the Real number system)",
+  "type": "number"
+}
+```
+
+#### Boolean
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/boolean/v/1",
+  "title": "Boolean",
+  "description": "A True or False value",
+  "type": "boolean"
+}
+```
+
+#### Null
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/null/v/1",
+  "title": "Null",
+  "description": "A placeholder value representing 'nothing'",
+  "type": "null"
+}
+```
+
+#### Object
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/object/v/1",
+  "title": "Object",
+  "description": "A plain JSON object with no pre-defined structure",
+  "type": "object"
+}
+```
+
+#### Empty List
+
+```json
+{
+  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+  "kind": "dataType",
+  "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/empty-list/v/1",
+  "title": "Empty List",
+  "description": "An Empty List",
+  "type": "array",
+  "const": []
+}
+```
+
+### <Hidden>The Type System </Hidden>Property Types
+
+A **property type** is the definition of the possible values that can be associated with a **property**.
+When represented in JSON schema, a property type satisfies the following meta-schema:
+
+<PropertyTypeMetaSchema />
+
+### <Hidden>The Type System </Hidden>Entity Types
+
+An **entity type** is the definition of the structure of an **entity**'s **properties**, and its outgoing **link entities** and their endpoint entities.
```

Comment:
for clarity, to avoid implying that an entity type defines the structure of its outgoing link entities
```suggestion
An **entity type** is the definition of (1) the structure of an **entity**'s **properties**, and (2) its outgoing **link entities** and their endpoint entities.
```